Crane/app/src/main/java/androidx/compose/samples/crane/base/CraneDrawer.kt

```diff
@@ -54,58 +54,74 @@ fun CraneDrawer() {
     ModalNavigationDrawer(
         drawerState = drawerState,
         drawerContent = {
-            ModalDrawerSheet {
-                screens.forEach { item ->
-                    NavigationDrawerItem(
-                        label = { Text(text = item) },
-                        selected = item == selectedItem.value,
-                        onClick = {
-                            scope.launch {
-                                drawerState.close()
-                            }
-                            selectedItem.value = item
-                        },
-                        modifier = Modifier
-                            .padding(
-                                paddingValues = NavigationDrawerItemDefaults.ItemPadding,
-                            ),
-                    )
+            CraneDrawerContent(
```

Comment:
This is not a slot API, you just extracted the drawerContent. Here is an example of how to create a slot API and what is its benefits.

Reply:
@majiddaeinejad We don't need to do that, IMO. The article is about **limiting** the **flexibility** of slot API which could be useful when creating custom/reusable components but it's not the case here. Moreover, we prefer to have that flexibility because we don't have any restrictions regarding the design and our limitation could be out of date soon, so it seems overengineering.

If I'm wrong, we can discuss it if you can come up with a code implementation to make it more specific.

Reply:
You're right, If you don't need to reuse it in other part of code then your approach is correct otherwise you can use slot API to make your composable function like a component to reuse it.